envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdater.kt

```diff
@@ -69,7 +73,8 @@ class SnapshotUpdater(
                             xdsSnapshot = newUpdate.xdsSnapshot ?: previous.xdsSnapshot
                     )
                 }
-                .doOnNext { result ->
+                // concat map guarantees sequential processing (unlike flatMap)
+                .concatMap { result ->
```

Comment:
As you can see, this PR changes implementation quite a bit even if parallel flag is set to false. Do you think I should create additional flag to allow reverting to old flow entirely? It would require a lot semi-duplicated code for "old flow" branch.